### PR TITLE
Add logging and asserts to update function

### PR DESCRIPTION
**Pull request by Aak242**

### CODE REVIEW

File: src/main/java/data/TaskManager.java [hunk 1]
```diff
@@ -62,10 +67,15 @@ public static void updateTask(LocalDate date, int taskIndex, String newTaskDescr
...
-            }
+            assert dayTasks != null;
+            assert taskIndexExists;
+
+            String oldDescription = dayTasks.get(taskIndex).getName();
+
+            Task task = new Task(newTaskDescription);
+            dayTasks.set(taskIndex, task);
+
+            logger.log(Level.INFO, "Updating task description from " + oldDescription + " to: " + newTaskDescription);
```
Comment (Mahesh1772) [inline src/main/java/data/TaskManager.java:78]: Nice job adding assertions

File: src/main/java/data/TaskManager.java [hunk 2]
```diff
@@ -18,10 +18,15 @@
 import static data.TaskManagerException.checkIfDateInCurrentMonth;
 import static storage.Storage.saveTasksToFile;
 
+import java.util.logging.Level;
```
Comment (Mahesh1772) [inline src/main/java/data/TaskManager.java:21]: Looks good to me

**Review by Mahesh1772 — COMMENTED** [review]
LGTM!

File: src/main/java/data/TaskManager.java (hunk 2 above)
Comment (clarencepohh) [inline src/main/java/data/TaskManager.java:21]: Might want to implement the global logger implemented by Nicholas

**Pull request merged**